### DISCUSSION TITLE
Record bufferSize removed?

### DISCUSSION THREAD

**Discussion by joewoodhouse** [katspaugh/wavesurfer.js#3292] [Q&A]
Hi all,

I'm new to this project and trying to get a real-time visualisation of some microphone audio working. I can see the example at https://wavesurfer.xyz/examples/?record.js which is great but also visualises my problem - the real-time waveform is flying from right to left incredibly fast, it's barely possible to see anything. I want to slow this down and have a buffer.

In doing some research I think the old "Microphone" plugin which was replaced by the recording plugin had a `bufferSize` option which from reading the old docs looks to be what I'm after. But there is no equivalent in the latest version.

Basically I'd like to maintain a couple of seconds buffer in the waveform when viewing the the real-time audio from the microphone - is there a way to do this? Any help would be much appreciated!

**Comment by katspaugh** [comment]
It might sound like what you want, but bufferSize is completely unrelated to bufferring in this case.
https://codepen.io/entonbiba/pen/ZKRgvZ – the old plugin behaved very similar to the new one in this regard.

**Reply by joewoodhouse** [reply]
@katspaugh  Great thanks for the info. So is there a way to achieve what I want using the latest version?

**Reply by katspaugh** [reply]
You could try recording short audios every 1-2 seconds and displaying them instead of the mic preview. You can hide the mic preview waveform with `height: 0`.

Something along these lines:
```js
// Mic preview
const wavesurfer = WaveSurfer.create({
  container: '#mic',
  waveColor: 'rgb(200, 0, 200)',
  progressColor: 'rgb(100, 0, 100)',
  height: 0
})

// Initialize the Record plugin
const record = wavesurfer.registerPlugin(RecordPlugin.create())


// Record every 1 second in a loop
let prevWavesurfer
record.on('record-end', (blob) => {
  if (prevWavesurfer) prevWavesurfer.destroy()

  // Render the buffer
  prevWavesurfer = WaveSurfer.create({
    container: document.body,
    waveColor: 'rgb(200, 100, 0)',
    progressColor: 'rgb(100, 50, 0)',
    url: URL.createObjectURL(blob)
  })

  // Start a new recording
  record.startRecording()
  setTimeout(() => {
    record.stopRecording()
  }, 1000)
})
```
